From 8f132536729069b9834910f1ca04dcff556b9096 Mon Sep 17 00:00:00 2001
From: Rapaka Naveen <naveen.rapaka@intel.com>
Date: Wed, 6 May 2015 00:49:39 +0530
Subject: [PATCH] EM/charger: BQ charger uses uninitialized variable

bq24192_probe in probe uses dev even if not initialized when
the argument id is not NULL

Change-Id: I0cd09433e917dd0982e83d860f8184cfe72a1b58
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1547
Signed-off-by: Rapaka Naveen <naveen.rapaka@intel.com>
---
 drivers/power/bq24192_charger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index 32c00eac5a2b..a16d73c7dcac 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -2135,11 +2135,11 @@ static int bq24192_probe(struct i2c_client *client,
 	}
 
 	chip->client = client;
+	dev = &client->dev;
 	if (id) {
 		chip->pdata = (struct bq24192_platform_data *)id->driver_data;
 	} else {
 #ifdef CONFIG_ACPI
-		dev = &client->dev;
 		if (!ACPI_HANDLE(dev)) {
 			i2c_set_clientdata(client, NULL);
 			kfree(chip);
-- 
1.9.1

